modules/SofaGeneralEngine/src/SofaGeneralEngine/ExtrudeQuadsAndGenerateHexas.inl

```diff
@@ -170,9 +171,16 @@ void ExtrudeQuadsAndGenerateHexas<DataTypes>::doUpdate()
     {
         for (int n=0; n<nSlices; n++)
         {
-            BaseMeshTopology::Hexa hexa = BaseMeshTopology::Hexa(surfaceQuads[q][3]*(nSlices+1)+n,   surfaceQuads[q][2]*(nSlices+1)+n,   surfaceQuads[q][1]*(nSlices+1)+n,   surfaceQuads[q][0]*(nSlices+1)+n,
-                    surfaceQuads[q][3]*(nSlices+1)+n+1, surfaceQuads[q][2]*(nSlices+1)+n+1, surfaceQuads[q][1]*(nSlices+1)+n+1, surfaceQuads[q][0]*(nSlices+1)+n+1);
-            extrudedHexas->push_back(hexa);
+            if (f_flipNormals.getValue())
```

Comment:
I'm not 100% it still make sense with latest version of c++ and compilators but I would put the if outside of the for loop to have:
```
if ()

   for

else

   for
```

Reply:
Sure. Thanks.